test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -4577,6 +4577,42 @@ BOOST_AUTO_TEST_CASE(constructing_enums_from_ints)
 	ABI_CHECK(callContractFunction("test()"), encodeArgs(1));
 }
 
+BOOST_AUTO_TEST_CASE(enum_referencing)
+{
+	char const* sourceCode = R"(
+			interface I {
+				enum Direction { Left, Right }
```

Comment:
Perhaps it would be better to use `enum Direction { A, B, Left, Right}` so that the return values show whether the interface or the library enum was used.